### PR TITLE
Fixes to the protocol

### PR DESCRIPTION
**Pull request by SFulpius**
Solves issues discussed in #97.
Also modified hashes/signatures so they correspond to what was discussed [here](https://docs.google.com/document/d/125nDMc2rc4L6pQgC92rC4X42-BOG8YCFcyzHeY-zY9A/edit) (see picture).

### CODE REVIEW

**Comment by RaulinN** [comment]
Hey @SFulpius! Thanks for doing these changes! :) 

I took the liberty to add a commit to your PR to correct "JsonSchema syntax". The two main errors I corrected among all files were:

1. you need to specify which properties are required in the `require` field. Otherwise, a JsonObject without the specific property is optional (I added that for the signature objects and removed the last_modified which don't exist anymore)
2. In an object definition, always add the "additionalProperties: false", otherwise, a JsonObject with $n$ additional unrelated properties would also pass the test

**Review by RaulinN — APPROVED** [review]

**Comment by SFulpius** [comment]
Thank you for the chanes @RaulinN.
I've also updated the signatures in the messageGeneral.json file.

File: query/method/message/messageGeneral.json [hunk 1]
```diff
@@ -31,16 +31,29 @@
...
+					},
+					"signature": {
+						"description": "[Base64String] witness' signature : Sign(message_id)",
+						"type": "string",
+						"contentEncoding": "base64",
+						"$comment": "Note: the strings are encoded in Base64"
+					}
+				},
+				"additionalProperties": false,
+				"required": ["witness", "signature"]
```
Comment (wouewouel) [inline query/method/message/messageGeneral.json:53]: Looks really fine to me, it resolves the problem of linking

**Review by wouewouel — APPROVED** [review]
The proposed solution looks fine to me as we no longer have the problem of remembering which witness signed the data.

**Pull request merged**